orangecontrib/timeseries/timeseries.py

```diff
@@ -19,16 +19,117 @@ def __init__(self, *args, **kwargs):
         self._interp_method = 'linear'
         self._interp_multivariate = False
         self._time_delta = None
-        # Set default time variable to first TimeVariable
+
+    @classmethod
+    def from_data_table(cls, table, detect_time_variable=False):
+        if isinstance(table, Timeseries) \
+                and table.time_variable is not None \
+                and table.time_delta is not None:
+            return table
+
+        # Set default time variable to first TimeVariable in:
+        search = table.domain.attributes + table.domain.metas
+
+        # Is there a time variable we can use?
         try:
-            self.time_variable = next(var for var in self.domain.attributes + self.domain.metas
-                                      if isinstance(var, TimeVariable))
+            time_variable = next(var for var in search
+                                 if var.is_time)
+            ts = super(Timeseries, cls).from_table(table.domain, table)
+            ts.time_variable = time_variable
+            return ts
         except (StopIteration, AttributeError):
             pass
 
+        if not detect_time_variable:
```

Comment:
So if detect_time_variable is False, we still look for appropriate tv above?

Reply:
Yeah, this behavior was moved from `__init__`

Reply:
👍